### ISSUE TITLE
SSHTunnelForwarder.daemon_forward_servers is not respected

### ISSUE THREAD

**Issue by bsamuel-ui** [pahaz/sshtunnel#102]
Opening this issue to get any thoughts before I do a PR.

Test case:

In ipython, construct a tunnel using:
```
tun = SSHTunnelForwarder((ssh_host, 22), ssh_username=user,
                              remote_bind_address=remote, local_bind_address=('127.0.0.1',),
                              ssh_pkey=ssh_key)
tun.daemon_forward_servers = True
tun.daemon_transport = True
tun.start()

In [12]: [(repr(t), t.isDaemon()) for t in threading.enumerate()]
Out[12]:
[('<_MainThread(MainThread, started 140736862168000)>', False),
 ('<HistorySavingThread(IPythonHistorySavingThread, started 123145316188160)>',
  True),
 ('<paramiko.Transport at 0xf68c048 (cipher aes128-ctr, 128 bits) (active; 1 open channel(s))>',
  True),
 ('<Thread(Srv-61825, started daemon 123145326698496)>', True),
 ('<Thread(Thread-282, started daemon 123145337208832)>', False)]
```

That last one should also be a daemon. The workaround is to set `_ThreadingForwardServer.daemon_threads=True`.

Questions:

* Is there a particular reason to have a global `DAEMON`?
  * It looks like public interface, but you can't update it once classes are initialized.
  * And, yuck, globals.
* Is there any benefit to having a separate `daemon_forward_servers` and `daemon_transport` field?

Proposal:
1. replace `DAEMON` with literal `False`
  * Still want a default to protect from crashes.
2. `_make_ssh_forward_server` sets `ssh_forward_server.daemon_threads = self.daemon_forward_servers`
3. Replace `SSHTunnelForwarder.daemon_*` with `.daemon`.

**Comment by fernandezcuesta** [comment]
PR #106 Fixed #102

**Comment by bsamuel-ui** [comment]
Thanks for merging!
